### PR TITLE
WIP: No longer need to build 'whisk/*' runtimes for builtins.

### PR DESCRIPTION
**Pull request by jonpspri**
Adjust the build and invoker to pull and deploy runtimes based on their fully-qualified Docker image names.  Runtimes documented in the Runtime Manifest file upon invoker start will be used for default implementations.  Overall, this removes some dependencies between `invoker-openwhisk` and the various runtimes and paves the way for a more dynamic approach to default runtimes.

(It also makes life easier as we look to convert from `distDocker` to `docker-gradle-plugin` for builds)

Due diligence continues but I'd like some additional eyes (and Travis runs) on this now in case I missed a glaring issue.

@csantanapr @markusthoemmes  Review and PG requested.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [X] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [X] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [X] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [X] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [ ] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [X] My changes _may_ require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: actionRuntimes/actionProxy/invoke.py [hunk 1]
```diff
@@ -1,136 +0,0 @@
-#!/usr/bin/env python
```
Comment (rabbah) [inline actionRuntimes/actionProxy/invoke.py]: this script is linked to from a few blogs and is handy for testing containers directly.

File: actionRuntimes/actionProxy/README.md [hunk 2]
```diff
@@ -1,37 +0,0 @@
-Skeleton for "docker actions"
```
Comment (rabbah) [inline actionRuntimes/actionProxy/README.md]: i'd like to suggest keeping this doc file and perhaps elaborating on it further as it describes how you add a new runtime container.

**Comment by markusthoemmes** [comment]
I would split this in at least two PRs. Removing the actionRuntimes from the repo is imho not part of the scope here. Ripping local images out of the invoker entirely, I think, is enough of a change for one PR.

File: ansible/group_vars/all [hunk 3]
```diff
@@ -207,6 +207,9 @@ docker:
   image:
     prefix: "{{ docker_image_prefix | default('whisk') }}"
     tag: "{{ docker_image_tag | default('latest') }}"
+  runtimes:
+    prefix: "{{ runtime_image_prefix | default('openwhisk') }}"
```
Comment (rabbah) [inline ansible/group_vars/all]: isn't this already defined somewhere else for the runtime manifest (i'll look when I have a chance).

File: actionRuntimes/actionProxy/invoke.py (hunk 1 above)
Comment (jonpspri) [reply actionRuntimes/actionProxy/invoke.py]: Okay I'll find another home for it.

File: actionRuntimes/actionProxy/README.md (hunk 2 above)
Comment (jonpspri) [reply actionRuntimes/actionProxy/README.md]: Makes sense.

File: ansible/group_vars/all (hunk 3 above)
Comment (jonpspri) [reply ansible/group_vars/all]: Runtime manifest uses docker_image_prefix -- I wanted to avoid the overloading.

**Comment by jonpspri** [comment]
Agreed this is pretty massive.  I'll see what I can do to break it up as it leaves WIP; some of the changes can be added now.  The reason for deleting the runtimes was to be able to run a 'clean' test of the other changes to ensure they weren't using some leftover pollution.

File: ansible/group_vars/all (hunk 3 above)
Comment (dgrove-oss) [reply ansible/group_vars/all]: @rabbah -- this what you were thinking of https://github.com/apache/incubator-openwhisk/blob/f3274d2155036dbd025902353f7b77c076ad1f86/ansible/group_vars/all#L203-#L210

Comment (jonpspri) [reply ansible/group_vars/all]: Yep.  Now we're going to differentiate between what's built locally by `./gradlew distDocker` (whisk/invoker, whisk/controller) and what's available for download as a "builtin" action `openwhisk/...`.  Of course, we should probably put some structure (tagged semver releases?) around the action runtimes in DockerHub.

**Comment by csantanapr** [comment]
I wan to see the flexibility of defining in the manifest json the runtimes eaxh can come different users/prefix for example “openwhisk” or “ibmfunctions” for any runtime and tag can be specified for each this metada is also already expose on the root of the api/v1/

**Comment by jonpspri** [comment]
@csantanapr The capability of putting prefix and tag are already in the manifest.json file.  We just need to get to the point where 'invoker' is not assuming `whick/` prefix, which should be happening in step 2 of the incremental rollout.  I'll be sure to highlight it when the PR comes out :)

**Comment by rabbah** [comment]
This is suprecded by #3745. @jonpspri my apologies I didn't realize you had this wip.

**Pull request closed without merging**